src/evaluate/evaluate.ts

```diff
@@ -45,28 +45,31 @@ export class Evaluator {
   constructor(public readonly context: EvaluationContext) {}
 
   public evaluateTemplate() {
+    this.evaluateMetadata();
     this.evaluateParameters();
+    this.evaluateRules();
     this.evaluateMappings();
     this.evaluateConditions();
+    this.evaluateTransform();
     this.evaluateResources();
     this.evaluateOutputs();
-    this.evaluateTransform();
-    this.evaluateMetadata();
-    this.evaluateRules();
   }
 
   private evaluateMappings() {
-    this.context.template.mappings.forEach(
-      (mapping, mapName) =>
-        new cdk.CfnMapping(this.context.stack, mapName, {
-          mapping: mapping.toObject(),
-        })
+    const scope = new Construct(this.context.stack, '$Mappings');
+    this.context.template.mappings.forEach((mapping, mapName) =>
+      new cdk.CfnMapping(scope, mapName, {
+        mapping: mapping.toObject(),
+      }).overrideLogicalId(mapName)
     );
   }
 
   private evaluateParameters() {
+    const scope = new Construct(this.context.stack, '$Parameters');
```

Comment:
Ha! Nice trick.

Reply:
Thanks, but it's stolen from here: https://github.com/aws/aws-cdk/blob/af1fb7ce4468b14aecbf24e2cc1c6efeb29761b7/packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts#L136